ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java

```diff
@@ -491,7 +499,7 @@ protected File writeToFile(String source) throws IOException {
      * @return File to use.
      * @throws IOException If file open failed.
      */
-    private File getBufferFile() throws IOException {
+    public File getBufferFile() throws IOException {
         if (this.bufferFile == null) {
             this.bufferFile = File.createTempFile(TEMP_FILE_PREFIX, TEMP_FILE_SUFFIX);
             addDebugDiagnostic("Using temp file: " + bufferFile.getAbsolutePath());
```

Comment:
Can you mention the reason to make this public ?

Reply:
We are using the content temp file created by shell to get completions, so in order to make that file content available for shell wrapper we have made it public.

Details on implementation of code completions: https://github.com/wso2/ballerina-plugin-vscode/issues/183#issuecomment-1080775556